src/cryptoadvance/specter/wallet.py

```diff
@@ -915,7 +919,14 @@ def check_utxo(self):
                                 utxo_item["txid"] == utxo_txid
                                 and utxo_item["vout"] == utxo_vout
                             ):
-                                tx_copy["amount"] = utxo_item["amount"]
+                                if utxo_item.get("amount"):
+                                    amount = utxo_item.get("amount")
+                                else:
+                                    # in the case of locked amounts, the stupid listlockunspent call does not contain reasonable utxo-data
+                                    amount = self.rpc.gettransaction(tx_copy["txid"])[
+                                        "details"
+                                    ][utxo_item["vout"]]["amount"]
+                                tx_copy["amount"] = amount
```

Comment:
We also need to set the address of this "utxo" as the address which is already set is coming from the tx-context and might not be the right one.